src/sidebar/services/groups.js

```diff
@@ -137,9 +137,40 @@ function groups($rootScope, store, api, isSidebar, localStorage, serviceUrl, ses
         params.document_uri = uri;
       }
       documentUri = uri;
-
-      // Fetch groups from the API.
-      return api.groups.list(params, null, { includeMetadata: true });
+      
+      if (features.flagEnabled('community_groups')) {
+        return $q.all([api.profile.groups.read({expand: "organization"}),
+               api.groups.list(params, null, { includeMetadata: true })]
+        ).then(data => {
+          const myGroups = data[0];
+          const listGroups = data[1].data;
+          let groups = [];
+          var i;
+  
+          const myGroupIds = [];
```

Comment:
No one beats MDN for JS reference IMHO. Their [page on Arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) has documentation of most of the things that are used above.

Reply:
BTW, the above is pseudocode. I haven't tried it.

Reply:
The reason I elected to use the loops here is if I didn't, it would add two extra loops over the entire array so I implemented it as raw loops for performance reasons. The array functions like find and map as I understand them still loop over the array so it seemed more appropriate to implement the simple for loops instead. I realize that potentially makes it less readable tho so if we feel like we're willing to sacrifice performance for readability I could be swayed in the other direction.